app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    if params[:search] == "pet" && params[:pet_name].present?
+      search_query = params[:pet_name]
+      @searched_pets = Pet.where("name ILIKE ?", "%#{search_query}%")
```

Comment:
This is logic we'd prefer to keep encapsulated at the model level. Also, take a look at the ApplicationRecord file -- this model method is already created for you! Since every model method inherits from ApplicationRecord, the `.search` method is available for you in every model! The beauty of a brownfield project is finding things you can use that other devs wrote